stock_orderpoint_route/models/stock_warehouse_orderpoint.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = "stock.warehouse.orderpoint"
+
+    route_ids = fields.Many2many(
+        "stock.location.route", string="Allowed routes", compute="_compute_route_ids"
+    )
+    route_id = fields.Many2one(
+        "stock.location.route",
+        string="Route",
+        domain="[('id', 'in', route_ids)]",
+        ondelete="restrict",
+    )
+
+    @api.depends("product_id", "warehouse_id", "warehouse_id.route_ids", "location_id")
+    def _compute_route_ids(self):
+        route_obj = self.env["stock.location.route"]
+        for wh in self.mapped("warehouse_id"):
+            wh_routes = wh.route_ids
+            for record in self.filtered(lambda r: r.warehouse_id == wh):
```

Comment:
can we avoid double loop?
can we avoid browsing?

Reply:
I ported the module as is, note: the number of warehouses is generally very low, shouldn't have too much an impact

Reply:
Fixed, we have less queries with a single, simple loop.